src/ra_server.erl

```diff
@@ -1290,6 +1304,13 @@ follower_catchup_cond(_, _Msg, State) ->
 wal_down_condition(_Msg, #{log := Log} = State) ->
     {ra_log:can_write(Log), State}.
 
+transfer_leadership_condition(#append_entries_rpc{term = Term},
```

Comment:
It is theoretically possible that the next message could be an `install_snapshot_rpc` so we'd need to handle this as well.